server-apollo/index.js

```diff
@@ -99,43 +114,47 @@ Query: {
 
 Mutation: {
     createUser: (obj, args, context, info) => {
-    args.id = new Date().getTime()
-    users.push(args)
-    return args
+        args.id = new Date().getTime()
+        args.version = 1
+        users.push(args)
+        return args
     },
     updateUser: (obj, args, context, info) => {
-    for (let user of users) {
-        if (args.id == user.id) {
-        user.name = args.name
-        user.feedback = args.feedback
-        user.dateOfBirth = args.dateOfBirth
-        return user
+        for (let user of users) {
+            if (args.id == user.id) {
+                if(conflictDetected(user.version, args.version)){
```

Comment:
Let's throw some serious logging here so we could showcase this and see conflicts without debugging.